metadata/src/from_into/v14.rs

```diff
@@ -277,36 +276,33 @@ impl ExtrinsicPartTypeIds {
             return Err(TryFromError::TypeNotFound(extrinsic_id));
         };
 
-        let params: HashMap<_, _> = extrinsic_ty
-            .type_params
-            .iter()
-            .map(|ty_param| {
-                let Some(ty) = ty_param.ty else {
-                    return Err(TryFromError::TypeNameNotFound(ty_param.name.clone()));
-                };
-
-                Ok((ty_param.name.as_str(), ty.id))
-            })
-            .collect::<Result<_, _>>()?;
+        let find_param = |name: &'static str| -> Option<u32> {
```

Comment:
Wo you know how big the extrinsic type params list is? 

We will do a linear search 4 times on instead 4 HashMap lookups which could be slower...

Reply:
Ok, I checked it myself it's `extrinsic_ty.len() == 4`